crates/env/src/backend.rs

```diff
@@ -162,9 +162,19 @@ impl CallFlags {
 
 /// Environmental contract functionality that does not require `Environment`.
 pub trait EnvBackend {
+    /// Writes the value to the contract storage under the given key.
+    ///
+    /// # Note
+    ///
+    /// This is an equivalent to the new [`set_contract_storage`][`Self::set_contract_storage`] method,
+    /// but in order to maintain legacy behavior it returns nothing.
```

Comment:
We should have this as part of the new functions, not the old one. We don't need to mention "legacy" behaviour or anything like that since it's part of the current release (and thus the current expected behaviour)